### ISSUE TITLE
Settings for BigFloat

### ISSUE THREAD

**Issue by maxchendt** [oxfordcontrol/Clarabel.jl#105]
```Julia
using Clarabel
CSf = Clarabel.Settings()
CSb = Clarabel.Settings{BigFloat}()
println(CSf.tol_feas,"\n" ,CSb.tol_feas) 
```

It seems that the Settings for `BigFloat ` are the same as those (tolerances etc) of `Float64`.

What sets of  settings for `BigFloat `  do you recommend?  We need a stringent one as we have more digits involved

**Comment by goulart-paul** [comment]
You are correct that the solver uses the same default settings regardless of the precision of the underlying data type.   It is  difficult to give numbers that will work in all cases, particularly given that the Julia BigFloat has user-definable precision.

That said, I can at least make some suggestions.   All of the parameters are listed here:  https://oxfordcontrol.github.io/ClarabelDocs/stable/api_settings/

As a starting point, I would suggest setting all of the parameters with names like `tol_*` to some tighter tolerance.   What that would be is up to you of course, but maybe start with 10^-12 in the absence of any other information.   The only one you may wish to leave the same would be `tol_ktratio`, which could either stay at 10^-7, or at least kept at a lower tolerance than the others.

A similar comment should hold for the `reduced_tol_*` parameter set.   Those are only used when the solver can't reach the desired precision anyway, but probably reasonably to set them to something like 10*(sqrt(X)), where X is whatever you have for used the full accuracy settings.   I suggest that only because it is roughly what we have there already.

I would start like that.   If you don't get good convergence, the next tolerance to tighten would be `iterative_refinement_reltol` and `iterative_refinement_abstol`.   Make this something like X*1e-4, i.e. try to maintain the same relative sizes as in the default settings.

Unless you get really weird behaviour for your problem, or your problem is very badly scaled or something, I would be inclined to leave everything else as it is.

You didn't ask this, but for anyone finding this in the future: for Float32 types, a reasonable starting point would be to take the square root of the default values for all parameters named above, leaving everything else unchanged.

**Comment by araujoms** [comment]
I tried that, didn't work, neither for `Float32` nor `BigFloat`. It gave sensible results for `Double64` and `Float128`. Perhaps you could simply copy the default tolerances from Hypatia? It works with all types I tried. They are set here: https://github.com/jump-dev/Hypatia.jl/blob/c27ef0d1eb9cb872a265437c6d96fcfa57369809/src/Solvers/Solvers.jl#L222-L249

To be more precise, I tested Clarabel and Hypatia with the following code:
```julia
using LinearAlgebra
using DoubleFloats
using Quadmath
using JuMP
import Hypatia
import Clarabel
import Random

function random_sym(::Type{T}, d) where {T <: AbstractFloat}
    x = randn(T, d, d)
    return Symmetric(x + x')
end

function mineig_sdp(x::Symmetric{T}, solver) where {T}
    d = size(x, 1)
    model = GenericModel{T}()
    @variable(model, λ)
    @constraint(model, x - λ * Symmetric(I(d)) ∈ PSDCone())
    @objective(model, Max, λ)
    set_optimizer(model, solver.Optimizer{T})
    if solver == Clarabel
        set_optimizer_attribute(model, "tol_gap_abs", sqrt(eps(T)))
        set_optimizer_attribute(model, "tol_gap_rel", sqrt(eps(T)))
        set_optimizer_attribute(model, "tol_feas", sqrt(eps(T)))
        set_optimizer_attribute(model, "tol_infeas_abs", sqrt(eps(T)))
        set_optimizer_attribute(model, "tol_infeas_rel", sqrt(eps(T)))
        set_optimizer_attribute(model, "tol_ktratio", eps(T)^T(0.35))
        set_optimizer_attribute(model, "iterative_refinement_reltol", eps(T)^T(0.75))
        set_optimizer_attribute(model, "iterative_refinement_abstol", eps(T)^T(0.75))
    end
    optimize!(model)
    return objective_value(model)
end

function sdp_error(::Type{T}, solver) where {T <: AbstractFloat}
    Random.seed!(1337)
    x = random_sym(T, 10)
    λ_sdp = mineig_sdp(x, solver)
    λ_analytic = minimum(eigvals(x))
    return abs(λ_sdp - λ_analytic)
end

function test_solver(solver)
    types = (Float32, Float64, Double64, Float128, BigFloat)
    error_vec = zeros(length(types))
    for i ∈ 1:length(types)
        error_vec[i] = sdp_error(types[i], solver)
    end
    return error_vec
end
```
Then `test_solver(Hypatia)` gives `[7.0e-04, 1.9e-08, 1.7e-13, 5.5e-15, 1.1e-31]`, and `test_solver(Clarabel)` gives `[4.7e+00, 1.1e-09, 1.5e-17, 8.5e-17, 8.0e-17]`.
